src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDirectoryEntry.java

```diff
@@ -156,4 +158,65 @@ public String toString() {
         .append("size", size)
         .toString();
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    FireStoreDirectoryEntry that = (FireStoreDirectoryEntry) o;
+    return Objects.equals(fileId, that.fileId)
+        && Objects.equals(isFileRef, that.isFileRef)
+        && Objects.equals(path, that.path)
+        && Objects.equals(name, that.name)
+        && Objects.equals(datasetId, that.datasetId)
+        && Objects.equals(fileCreatedDate, that.fileCreatedDate)
+        && Objects.equals(checksumCrc32c, that.checksumCrc32c)
+        && Objects.equals(checksumMd5, that.checksumMd5)
+        && Objects.equals(size, that.size);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(
+        fileId,
+        isFileRef,
+        path,
+        name,
+        datasetId,
+        fileCreatedDate,
+        checksumCrc32c,
+        checksumMd5,
+        size);
+  }
+
+  public static FireStoreDirectoryEntry fromTableEntity(TableEntity entity) {
+    return new FireStoreDirectoryEntry()
+        .fileId(entity.getProperty("fileId").toString())
```

Comment:
String property names could be final class fields since they're used in both serialization and deserialization